src/controllers/mainController.ts

```diff
@@ -1632,6 +1644,30 @@ export default class MainController implements vscode.Disposable {
         });
     }
 
+    public onToggleActualPlan(
+        isEnable: boolean,
+        callbackThis?: MainController,
+    ): void {
+        // the 'this' context is lost in retry callback, so capture it here
+        const self: MainController = callbackThis ? callbackThis : this;
+        const uri = self._vscodeWrapper.activeTextEditorUri;
+
+        if (isEnable && !this._actualPlanStatuses.includes(uri)) {
```

Comment:
so we only need self for the "vscodeWrapper" but the rest of the step is fine from whatever "this"?